### PR TITLE
Fix loop closure violations in unit tests

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

This PR fixes:
- loop closure violations in polybft unit tests and 
- makes `TestExtra_CreateValidatorSetDelta_Cases` runs deterministically (by fixing randomization of voting power updates)

# Changes include

- [x] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### CODE REVIEW

File: consensus/polybft/extra_test.go
Comment (ferranbt) [inline consensus/polybft/extra_test.go]: Is this an issue because the tests run in Parallel?

Comment (Stefan-Ethernal) [reply consensus/polybft/extra_test.go]: Yes and I fixed it because VSC reported violation (although in this case we have only single test case to run). However without having `t.Parallel()` I'd say sub-tests would be executed in parallel because of `t.Run` which spawns a go routine.

**Review by vcastellm — APPROVED** [review]
A bit weird because we only have 1 test case but LGTM if it fixes the issue

**Comment by Stefan-Ethernal** [comment]
> A bit weird because we only have 1 test case but LGTM if it fixes the issue

Let me give you a bit more context, because this PR fixes two things:
1. VS Code reported warning for loop closures and it perfectly makes sense, especially for `Test_buildLogsFromReceipts`, because there we have multiple test cases and tests weren't executed correctly without capturing a loop variable (although it is not easily noticeable, without adding logs to the test, because both test cases are running succesfully). For more info, read [this](https://www.gopherguides.com/articles/table-driven-testing-in-parallel#:~:text=gopherguides/training%20%20%20%20%20%20%20%200.156s-,Bug%20Time!,-How%20closely%20did).
On the other hand, it doesn't make any difference at the moment fixing it for `TestExtra_CreateValidatorSetDelta_Cases`, since currently there is only single sub test. Though we could easily decide to expand it in the future and we have landmine there and it is good to address it right now.
2. `TestExtra_CreateValidatorSetDelta_Cases` was flaky, because of assigning same random value for voting power as the old one (which is 1). That part was fixed here: https://github.com/0xPolygon/polygon-edge/pull/1053/files#diff-2f601f47fe8d02bd0d03ad2bbb4c245778e316affb10eb5497b9c4d1b58d11d1R393

File: consensus/polybft/extra_test.go
Comment (Stefan-Ethernal) [reply consensus/polybft/extra_test.go]: @ferranbt more detailed explanation [here](https://github.com/0xPolygon/polygon-edge/pull/1053#issuecomment-1353257905)

**Review by goran-ethernal — APPROVED** [review]

**Review by b00ris — APPROVED** [review]

**Pull request merged**
